src/server/botd-firewall/cloudflareApiHelper.ts

```diff
@@ -0,0 +1,67 @@
+import { getBlockedIps } from './blockedIpsDatabase';
+import { CloudflareRule, buildFirewallRules } from './buildFirewallRules';
+
+async function updateRulesetUsingCloudflareAPI(rules: CloudflareRule[]) {
+  const apiToken = process.env.CLOUDFLARE_API_TOKEN ?? '';
+  const zoneId = process.env.CLOUDFLARE_ZONE_ID ?? '';
+  // You can get your Cloudflare API token, and zone ID from your Cloudflare dashboard.
+  // But you might need to call the API to find the custom ruleset ID. See getCustomRulesetId() below.
+  const customRulesetId = process.env.CLOUDFLARE_RULESET_ID ?? '';
+
+  const url = `https://api.cloudflare.com/client/v4/zones/${zoneId}/rulesets/${customRulesetId}`;
+  const options = {
+    method: 'PUT',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${apiToken}`,
+    },
+    body: JSON.stringify({
+      description: 'Custom ruleset for blocking Fingerprint-detected bot IPs',
+      kind: 'root',
+      name: 'default',
+      phase: 'http_request_firewall_custom',
+      rules,
+    }),
+  };
+
+  const response = await fetch(url, options);
+  if (response.ok) {
+    return await response.json();
+  } else {
+    console.log(response.statusText, await response.json());
+    throw new Error('Updating firewall ruleset failed', { cause: response.statusText });
+  }
```

Comment:
Let's unwrap the if/else block.

Reply:
Exactly! 😁 And the same for other places

Reply:
Thanks, done!